pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -275,6 +281,16 @@ func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode C
 	return subsetClusters
 }
 
+func (cb *ClusterBuilder) applyMetadataExchange(c *cluster.Cluster) {
+	// We configure MX only if we have some Telemetry configured. This essentially acts as a feature flag
```

Comment:
does this comment still match the intent/code?  It seems like it is now just gated on the actual feature flag.

Reply:
Oops, will need to go through the code. I pulled directly from the original PR and hacked on it until it compiled - bad comments compile :slightly_smiling_face: